test/native/libs/vaddrdif.ld

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+SECTIONS
+{
+  . = ALIGN(0x1000);
+  .text 0x2000 : AT(0x1000)
```

Comment:
Thank you for adding the test, it's very helpful!
I'd make virtual load address completely different (e.g. `0xab000`) rather than 0x1000 increment to 0x1000 offset. Otherwise, since 0x1000 is overloaded with different meanings, it's easy to mix up things (I almost implemented a wrong fix that accidentally worked).